src/compliance.rs

```diff
@@ -12,19 +12,15 @@ use std::{
 use tokio::{spawn, sync::RwLock, task::JoinHandle, time::interval};
 use xml::reader::{EventReader, XmlEvent};
 
+#[derive(Default)]
 pub struct Compliance {
     sanctioned_addresses: Arc<RwLock<HashMap<String, bool>>>,
     last_update: Arc<RwLock<i64>>,
 }
 
-impl Default for Compliance {
-    fn default() -> Self {
-        Self::new()
-    }
-}
-
 impl Compliance {
     const BTC_ID: &'static str = "344";
+    const TICK_INTERVAL: u64 = 600;
```

Comment:
so every 10min, maybe we can relax this even more... mmm... I think it'd be a good idea to log it when we detect an update on the sanction list, so that we can have an idea of the frequency and revise this number later

Reply:
The good thing is that there is pre-fetch stage which will fetch a few bytes initially. Those few bytes are enough to see if there is a fresh version of the dataset. Basically, the XML includes a tag that has the publish date of the document. So if the there is no fresh data then we don't fetch and parse the large XML file. This takes a few milliseconds to run.